### PR TITLE
Added devise for user, devise views, property mvc, user dashboard

### PR DESCRIPTION
**Pull request by zstrad44**
Added a few different user stories in this commit, forgot to break it up.

**User stories:** 
- As a user, I need to create an account and be able to log in
- As a user, I need to add/ edit/ delete my properties in my account
- As a user, I need to add my properties to my account

**Objective of commit:** 
- Create basic design outline of app
- Add devise for user authentication
- Create custom devise views
- Create user dashboard
- Create MVC for property

### CODE REVIEW

File: app/assets/javascripts/properties.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (alexstophel) [inline app/assets/javascripts/properties.coffee]: You can ditch these `.coffee` files unless you're going to write CoffeeScript. If you ditch these files, feel free to also remove the gem from the Gemfile. :)

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -0,0 +1,204 @@
+/*
```
Comment (alexstophel) [inline app/assets/stylesheets/application.scss]: You can remove all of these comments now (lines 1-15).

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,3 +1,19 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :authenticate_user!
+
```
Comment (alexstophel) [inline app/controllers/application_controller.rb]: Extra newline here.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+  before_action :authenticate_user!
+  
+  
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:username, :email, :password ])
+    devise_parameter_sanitizer.permit(:account_update, keys: [:username, :email, :password, :current_password])
+    
+     devise_parameter_sanitizer.permit(:sign_in) do |user_params|
```
Comment (alexstophel) [inline app/controllers/application_controller.rb]: Tabbing is slightly off.

File: app/controllers/properties_controller.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  end
+
+  def edit
+    @property = Property.find(params[:id])
+  end
+
+  def create
+      @property = Property.new(property_params)
+      @property.user = current_user
+
```
Comment (alexstophel) [inline app/controllers/properties_controller.rb]: Extra newline

File: app/controllers/properties_controller.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  def new
+    @property = Property.new
+  end
+
+  def edit
+    @property = Property.find(params[:id])
+  end
+
+  def create
+      @property = Property.new(property_params)
```
Comment (alexstophel) [inline app/controllers/properties_controller.rb]: You might also get away with doing:

```
current_user.build_property(property_params)
```

[Check out this page in the docs](http://guides.rubyonrails.org/association_basics.html#has-one-association-reference) for more info!

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
+  # def index
```
Comment (alexstophel) [inline app/controllers/users_controller.rb]: Do you need this commented out code anymore?

File: Gemfile [hunk 8]
```diff
@@ -35,8 +35,8 @@ end
 
 gem 'sass-rails', '~> 5.0'
 gem 'uglifier', '>= 1.3.0'
-gem 'coffee-rails', '~> 4.2'
 gem 'turbolinks', '~> 5'
+gem 'coffee-rails', '~> 4.2'
```
Comment (zstrad44) [inline Gemfile]: I tried removing the coffee gem but got an error after I restarted the server, so I added it back in for now.

**Pull request merged**